From d3d963bc3a83a41cf42a50050abdbf3d6ff3f2fa Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Tue, 3 Mar 2020 23:14:50 +0800
Subject: [PATCH] drm/imx/imx-drm-core: fix imx_drm_bind function bind
 imx-lcdif-crtc fails

imx_drm_bind set imx sets drm-> mode_config.allow_fb_modifiers = true;
when bind imx-lcdif-crtc, it will perform lcdif_plane_init initialization
and find imx sets drm-> mode_config.allow_fb_modifiers = true will return error
this flag has been modified to be used as follows, for DPU and DCSS kms
    if (has_dpu (dev) || has_dcss (dev)) {
        drm-> mode_config.allow_fb_modifiers = true;
        dev_dbg (dev, "allow fb modifiers \ n");
    }

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 drivers/gpu/drm/imx/imx-drm-core.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/imx-drm-core.c b/drivers/gpu/drm/imx/imx-drm-core.c
index c6bcd917126f..0eafa6be2937 100644
--- a/drivers/gpu/drm/imx/imx-drm-core.c
+++ b/drivers/gpu/drm/imx/imx-drm-core.c
@@ -307,7 +307,6 @@ static int imx_drm_bind(struct device *dev)
 	drm->mode_config.min_height = 1;
 	drm->mode_config.max_width = 4096;
 	drm->mode_config.max_height = 4096;
-	drm->mode_config.allow_fb_modifiers = true;
 
 	if (has_dpu(dev) || has_dcss(dev)) {
 		drm->mode_config.allow_fb_modifiers = true;
-- 
2.17.1

